src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -26,27 +28,28 @@
 using Roslyn.Utilities;
 using static Microsoft.CodeAnalysis.CodeActions.CodeAction;
 using CodeFixGroupKey = System.Tuple<Microsoft.CodeAnalysis.Diagnostics.DiagnosticData, Microsoft.CodeAnalysis.CodeActions.CodeActionPriority>;
+using IUIThreadOperationContext = Microsoft.VisualStudio.Utilities.IUIThreadOperationContext;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.Suggestions
 {
     internal partial class SuggestedActionsSourceProvider
     {
-        private class SuggestedActionsSource : ForegroundThreadAffinitizedObject, ISuggestedActionsSource2
+        private class SuggestedActionsSource : ForegroundThreadAffinitizedObject, ISuggestedActionsSource3
         {
             private readonly ISuggestedActionCategoryRegistryService _suggestedActionCategoryRegistry;
 
             // state that will be only reset when source is disposed.
-            private SuggestedActionsSourceProvider _owner;
-            private ITextView _textView;
-            private ITextBuffer _subjectBuffer;
-            private WorkspaceRegistration _registration;
+            private SuggestedActionsSourceProvider? _owner;
+            private ITextView? _textView;
+            private ITextBuffer? _subjectBuffer;
+            private WorkspaceRegistration? _registration;
```

Comment:
💡 Either of the following would substantially clean up the code:

1. Don't set them to null on dispose
2. Provide an accessor for each field that returns non-null:

    ```csharp
    private SuggestedActionsSourceProvider Owner => _owner ?? throw new ObjectDisposedException();
    ```

Reply:
I instead added RoslynDebug.Assert based on the checks earlier in the code paths, this removed all unnecessary `?.` that I added.